targets/TARGET_NORDIC/TARGET_NRF5/i2c_api.c

```diff
@@ -371,17 +375,20 @@ int i2c_start(i2c_t *obj)
 int i2c_stop(i2c_t *obj)
 {
     NRF_TWI_Type *twi = m_twi_instances[TWI_IDX(obj)];
+    uint32_t t0;
 
     // The current transfer may be suspended (if it is RX), so it must be
     // resumed before the STOP task is triggered.
     nrf_twi_task_trigger(twi, NRF_TWI_TASK_RESUME);
     nrf_twi_task_trigger(twi, NRF_TWI_TASK_STOP);
-    uint32_t remaining_time = TIMEOUT_VALUE;
+
+    t0 = us_ticker_read();
```

Comment:
The upcoming microsecond ticker specification allows the microsecond ticker to use frequencies other than 1MHz, and bit widths fewer than 32 which could cause problems here.

To ensure the microsecond ticker is initialized, and to ensure the right frequency you might want to go through the mbed ticker api. It would look something like this ```ticker_read(get_us_ticker_data());```.